core/o11y/Cargo.toml

```diff
@@ -18,9 +18,9 @@ opentelemetry-jaeger = "0.16"
 thiserror = "1"
 tracing = { version = "0.1.13", features = ["std"] }
 tracing-appender = "0.2.2"
-tracing-opentelemetry = "0.17"
+tracing-opentelemetry = { version = "0.17", optional = true }
 tracing-serde = "0.1"
 tracing-subscriber = { version = "0.3.9", features = ["fmt", "env-filter", "std"] }
 
 [features]
-opentelemetry = []
+opentelemetry = ["dep:tracing-opentelemetry"]
```

Comment:
This seems to be pre-existing, but why we make this a compile-time feature? Seems simpler to just always compile telemetry in, and configure it with a runtime flag?